### ISSUE TITLE
Custom  Dataset

### ISSUE THREAD

**Issue by Zumbalamambo** [kirumang/Pix2Pose#12]
How to prepare the custom dataset?

**Comment by kirumang** [comment]
Hi,
The best way is to prepare the custom dataset in [the BOP format ](https://github.com/thodan/bop_toolkit/blob/master/docs/bop_datasets_format.md). 
The essential components are files that are 
- camera.json (camera intrinsic), 
- models (ply files and models_info.json)
- train (if training images exist) 
If you don't have training images, I would recommend using other rendering pipelines that use Blender instead of basic OpenGL based pipeline in this repository (e.g., [BlenderProc](https://github.com/DLR-RM/BlenderProc/blob/master/README_BlenderProc4BOP.md)

After that, you have to set up the directory and the name of the dataset in tools/bop_io.py properly.
```python3
elif(dataset=='dataset_name'):
        bop_dataset_dir = os.path.join(bop_dir,"dataset_dir")
        test_dir = bop_dataset_dir+"/test" #(optional)
        train_dir = bop_dataset_dir+"/train" 
        model_dir = bop_dataset_dir+"/models"+postfix_model
        model_scale=0.001 #scale of the 3D models
```
https://github.com/kirumang/Pix2Pose/blob/843effe0097e9982f4b07dd90b04ede2b9ee9294/tools/bop_io.py#L105

And, follow the steps including the training of a detection pipeline.

Further questions are more than welcome!
Thank you

**Comment by HamilkarBrq** [comment]
Hey @Zumbalamambo, @kirumang,
I am preparing a custom dataset with the BOP toolkit as well, which actually already worked out pretty well apart from centering the view of BlenderProc on the custom dataset objects. The rendered pictures I am currently able to create with the `bop_loader` function usually only contain small parts of the custom objects located in the bottom-right corner. Did you have similar problems and found a workaround or did you successfully manage to create your dataset?
So far I managed to:
- include the custom dataset in all necessary functions (that I was able to find) from `bop_toolkit` and `Pix2Pose` so it is recognized as a BOP dataset
- transform custom .ply models into colored xyz-models
- create annotated pictures in `/path/to/bop_data/custom_dataset` with ground truths, masks and models_info.json files

What I guess is still necessary to be done to train my own Pix2Pose weights:
- create training dataset with pictures of each individual model
- create scenes with custom objects and BOP dataset objects  

Please let me know if any of you sees an obvious flaw in my work so far or has an idea how I should proceed to manage creating a custom dataset. I already posted an issue in the BlenderProc repository ([Link](https://github.com/DLR-RM/BlenderProc/issues/143)), but couldn't find any solution so far, so any help is greatly appreciated.

**Edit:**
Fixing wrong values in the `camera.json` file did the magic for me.

Regards,
Jan
